src/configuration-factory.ts

```diff
@@ -1,15 +1,20 @@
 import {
   Flag,
   HybridConfigurationStore,
+  IAsyncStore,
   IConfigurationStore,
   MemoryOnlyConfigurationStore,
   MemoryStore,
 } from '@eppo/js-client-sdk-common';
 
 import { LocalStorageBackedAsyncStore } from './local-storage';
 
-export function configurationStorageFactory(): IConfigurationStore<Flag> {
-  if (hasWindowLocalStorage()) {
+export function configurationStorageFactory(
+  persistenceStore?: IAsyncStore<Flag>,
```

Comment:
Consider renaming to `persistentStore`, which I think is what this is usually called, unless I'm misunderstanding what this is

Reply:
And I think it makes sense as an argument in init rather than a `usePersistentStore(store)` method that could modify the client's storage after it has been initialized. If there's a scenario where swapping stores is useful, it doesn't seem like we're blocked from providing that method in the future.

Reply:
> And I think it makes sense as an argument in init rather than a usePersistentStore(store) method that could modify the client's storage after it has been initialized

That was my thinking too - can't imagine wanting to swap this out in the middle of a session.